editor/src/components/navigator/left-pane.tsx

```diff
@@ -763,6 +768,249 @@ const GithubSpinner = () => {
+        setUsersRepositories(repositories)
+      },
+      [setUsersRepositories],
+    )
+
+    React.useEffect(() => {
+      if (githubAuthenticated) {
+        void getUsersPublicGithubRepositories(dispatch, setUsersRepositoriesCallback)
+      }
+    }, [githubAuthenticated, dispatch, setUsersRepositoriesCallback])
+
+    const onInputChangeTargetRepository = React.useCallback(
+      (event: React.ChangeEvent<HTMLInputElement>) => {
+        setTargetRepository(event.currentTarget.value)
+      },
+      [setTargetRepository],
+    )
+
+    const filteredRepositories = React.useMemo(() => {
+      if (usersRepositories == null) {
+        return null
+      } else {
+        let filteredResult = usersRepositories.map((repository) => {
+          return {
+            ...repository,
+            importPermitted: true,
+          }
+        })
+        if (targetRepository != null) {
+          filteredResult = filteredResult.filter((repository) => {
+            return (
+              repository.fullName.includes(targetRepository) ||
+              repository.name?.includes(targetRepository)
+            )
+          })
+        }
+        return filteredResult
+      }
+    }, [usersRepositories, targetRepository])
+
+    const filteredRepositoriesWithSpecialCases = React.useMemo(() => {
+      if (filteredRepositories == null) {
+        return null
+      } else {
+        const parsedRepo =
+          targetRepository == null ? null : parseGithubProjectString(targetRepository)
+        if (parsedRepo == null) {
+          return filteredRepositories
+        } else {
+          const ownerRepo = `${parsedRepo.owner}/${parsedRepo.repository}`
+          const alreadyIncludesEntry =
+            filteredRepositories?.some((repo) => repo.fullName === ownerRepo) ?? false
+          if (alreadyIncludesEntry) {
+            return filteredRepositories
+          } else {
+            const additionalEntry: RepositoryRowProps = {
+              fullName: parsedRepo.repository,
+              avatarUrl: '',
+              private: true,
+              description: null,
+              name: null,
+              updatedAt: `${new Date()}`,
+              defaultBranch: 'main',
+              importPermitted: false,
+            }
+            return [...filteredRepositories, additionalEntry]
+          }
+        }
+      }
+    }, [filteredRepositories, targetRepository])
+
+    return (
+      <>
+        <UIGridRow padded variant={'<-------------1fr------------->'}>
+          <StringInput
+            placeholder={
+              filteredRepositoriesWithSpecialCases == null
+                ? 'Loading repositories...'
+                : 'owner/repository'
+            }
+            onChange={onInputChangeTargetRepository}
+            list={'repositories-list'}
+            id={'repositories-input'}
+            testId={'repositories-input'}
+            name={'repositories-input'}
+            value={targetRepository}
```

Comment:
watch out - this does not update when the repo is imported for the first time

Reply:
I smell a dependency being missed out.

Reply:
Fixed in 1a05bc536.